### ISSUE TITLE
Find a better way to generate salts and secrets

### ISSUE THREAD

**Issue by AntoineRondelet** [AntoineRondelet/snark-mixer#4]
Rather than generating salts and secrets randomly, if could probably be better for the client to be initialized with a random number, from which to start incrementing each salt for the nullifier (same for the secrets), to keep track/store all nullifiers and secrets more easily.

**Comment by HarryR** [comment]
If the secret and salt are incremented, then providing any secret to anybody else (as required by the protocol) would reveal all future secrets that the person will use, which is... possibly needs some re-thought?

**Comment by AntoineRondelet** [comment]
Yeah true... I need to put more thinking into it..! This is completely flawed indeed... this ticket reminds me to think about it

**Comment by barryWhiteHat** [comment]
One thing i want to add here is that if a user sends two deposits with the same nullifier then only a single payment can be withdrawn. This is a little annoying when i want to pay someone else. Its very unlikely it will get blocked but its still possible. So i thought about using an incremental nullifier here. But discarded it because of race conditions.
